contracts/hooks/FullRange.sol

```diff
@@ -0,0 +1,407 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import {IPoolManager} from "@uniswap/v4-core/contracts/interfaces/IPoolManager.sol";
+import {PoolManager} from "@uniswap/v4-core/contracts/PoolManager.sol";
+import {Pool} from "@uniswap/v4-core/contracts/libraries/Pool.sol";
+import {Hooks} from "@uniswap/v4-core/contracts/libraries/Hooks.sol";
+import {BaseHook} from "../BaseHook.sol";
+
+import {IHooks} from "@uniswap/v4-core/contracts/interfaces/IHooks.sol";
+import {CurrencyLibrary, Currency} from "@uniswap/v4-core/contracts/types/Currency.sol";
+import {TickMath} from "@uniswap/v4-core/contracts/libraries/TickMath.sol";
+import {BalanceDelta, BalanceDeltaLibrary} from "@uniswap/v4-core/contracts/types/BalanceDelta.sol";
+import {IERC20Minimal} from "@uniswap/v4-core/contracts/interfaces/external/IERC20Minimal.sol";
+import {ILockCallback} from "@uniswap/v4-core/contracts/interfaces/callback/ILockCallback.sol";
+import {PoolId, PoolIdLibrary} from "@uniswap/v4-core/contracts/types/PoolId.sol";
+import {PoolKey} from "@uniswap/v4-core/contracts/types/PoolKey.sol";
+import {FullMath} from "@uniswap/v4-core/contracts/libraries/FullMath.sol";
+import {UniswapV4ERC20} from "./UniswapV4ERC20.sol";
+import {Position} from "@uniswap/v4-core/contracts/libraries/Position.sol";
+import "@uniswap/v4-core/contracts/libraries/FixedPoint128.sol";
+import {FixedPoint96} from "@uniswap/v4-core/contracts/libraries/FixedPoint96.sol";
+import {FixedPointMathLib} from "solmate/utils/FixedPointMathLib.sol";
+import {ILockCallback} from "@uniswap/v4-core/contracts/interfaces/callback/ILockCallback.sol";
+
+import "forge-std/console2.sol";
+
+import "../libraries/LiquidityAmounts.sol";
+
+contract FullRange is BaseHook, ILockCallback {
+    using CurrencyLibrary for Currency;
+    using PoolIdLibrary for PoolKey;
+
+    /// @notice Thrown when trying to interact with a non-initialized pool
+    error PoolNotInitialized();
+
+    /// @dev Min tick for full range with tick spacing of 60
+    int24 internal constant MIN_TICK = -887220;
+    /// @dev Max tick for full range with tick spacing of 60
+    int24 internal constant MAX_TICK = -MIN_TICK;
+
+    uint256 public constant MINIMUM_LIQUIDITY = 10 ** 3;
```

Comment:
looks like we haven't integrated this yet

Reply:
i don't think we need this constant anymore!